ImportBlocks/ImportBlocks.cs

```diff
@@ -23,31 +23,91 @@ private bool OnExport(string[] args)
+                    return true;
+                }
+                path = $"root.{start}.acc";
+                WriteStateRoots(start, count, path);
+            }
+            else if (string.Equals(args[1], "block", StringComparison.OrdinalIgnoreCase)
+                || string.Equals(args[1], "blocks", StringComparison.OrdinalIgnoreCase))
+            {
+                if (args.Length >= 3 && uint.TryParse(args[2], out start))
+                {
+                    if (Blockchain.Singleton.Height < start) return true;
+                    count = args.Length >= 4 ? uint.Parse(args[3]) : uint.MaxValue;
+                    count = Math.Min(count, Blockchain.Singleton.Height - start + 1);
+                    path = $"chain.{start}.acc";
+                    writeStart = true;
+                }
+                else
+                {
+                    start = 0;
+                    count = Blockchain.Singleton.Height - start + 1;
+                    path = args.Length >= 3 ? args[2] : "chain.acc";
+                    writeStart = false;
+                }
+
+                WriteBlocks(start, count, path, writeStart);
             }
             else
             {
-                start = 0;
-                count = Blockchain.Singleton.Height - start + 1;
-                path = args.Length >= 3 ? args[2] : "chain.acc";
-                writeStart = false;
+                return false;
             }
 
-            WriteBlocks(start, count, path, writeStart);
-
             Console.WriteLine();
             return true;
         }
 
+        private void WriteStateRoots(uint start, uint count, string path)
+        {
+            uint end = start + count - 1;
+            using (FileStream fs = new FileStream(path, FileMode.OpenOrCreate, FileAccess.ReadWrite, FileShare.None, 4096, FileOptions.WriteThrough))
+            {
+                if (fs.Length > 0)
+                {
+                    byte[] buffer = new byte[sizeof(uint)];
+                    fs.Seek(sizeof(uint), SeekOrigin.Begin);
+                    fs.Read(buffer, 0, buffer.Length);
+                    start += BitConverter.ToUInt32(buffer, 0);
+                    fs.Seek(sizeof(uint), SeekOrigin.Begin);
+                }
+                else
+                {
+                    fs.Write(BitConverter.GetBytes(start), 0, sizeof(uint));
+                }
+                if (start <= end)
+                    fs.Write(BitConverter.GetBytes(count), 0, sizeof(uint));
+                fs.Seek(0, SeekOrigin.End);
+                for (uint i = start; i <= end; i++)
+                {
+                    StateRootState state = Blockchain.Singleton.Store.GetStateRoots().TryGet(i);
+                    byte[] array = state.StateRoot.ToArray();
+                    fs.Write(BitConverter.GetBytes(array.Length), 0, sizeof(int));
+                    fs.Write(array, 0, array.Length);
+                    Console.SetCursorPosition(0, Console.CursorTop);
```

Comment:
We should use the progress bar https://github.com/neo-project/neo-node/blob/master/neo-cli/CLI/ConsolePercent.cs

Reply:
> We should use the progress bar https://github.com/neo-project/neo-node/blob/master/neo-cli/CLI/ConsolePercent.cs

Maybe we should merge this first and pull another request for progress bar.